### PR TITLE
Supplier Relationship Management module: Initial implementation

### PR DESCRIPTION
**Pull request by santostelmo**
# Context
We want to be able to manage leads coming from suppliers. The flow is very similar to the existing one in CRM. Main change is that leads can be converted in purchases. 
What is done in this improvement is:
1. Reuse `lead.crm` model doing the distinction between a customer vs supplier lead
2. Adapt views
3. Generate purchase.orders from supplier leads

### CODE REVIEW

File: srm/models/crm_lead.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        order.company_id,
+                        order.date_order or fields.Date.today(),
+                    )
+            lead.purchase_amount_total = total
+            lead.request_for_quotation_count = rfq_cnt
+            lead.purchase_order_count = purchase_order_cnt
+
+    def _create_customer(self):
+        """It can be a customer or supplier depending on lead request type"""
+        self = self.with_context(res_partner_search_mode=self.request_type)
```
Comment (StephaneMangin) [inline srm/models/crm_lead.py:62]: Does it really needed for a customer creation ? It is mainly used to search partners and create account.

File: srm/views/srm_lead_views.xml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        <field name="action" ref="qoqa_srm.srm_opportunity_action_dashboard"/>
+    </record>
+
+    <record id="srm_menu_leads" model="ir.ui.menu">
+        <field name="action" ref="qoqa_srm.srm_lead_all_leads"/>
+    </record>
+
+    <record id="menu_srm_opportunities" model="ir.ui.menu">
+        <field name="action" ref="qoqa_srm.action_your_pipeline"/>
+    </record>
```
Comment (StephaneMangin) [inline srm/views/srm_lead_views.xml]:
```suggestion
    <record id="srm_menu_root" model="ir.ui.menu">
        <field name="action" ref="srm.srm_lead_action_pipeline"/>
    </record>

    <record id="srm_opportunity_report_menu_lead" model="ir.ui.menu">
        <field name="action" ref="srm.srm_lead_action_dashboard"/>
    </record>

    <record id="srm_opportunity_report_menu" model="ir.ui.menu">
        <field name="action" ref="srm.srm_opportunity_action_dashboard"/>
    </record>

    <record id="srm_menu_leads" model="ir.ui.menu">
        <field name="action" ref="srm.srm_lead_all_leads"/>
    </record>

    <record id="menu_srm_opportunities" model="ir.ui.menu">
        <field name="action" ref="srm.action_your_pipeline"/>
    </record>
```

File: srm/models/crm_lead.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        return super(CrmLead, self)._create_customer()
+
+    def action_lead_rfq_new(self):
+        if not self.partner_id:
+            return self.env["ir.actions.actions"]._for_xml_id(
+                "qoqa_srm.srm_rfq_partner_action"
```
Comment (StephaneMangin) [inline srm/models/crm_lead.py]:
```suggestion
                "srm.srm_rfq_partner_action"
```

File: srm/models/crm_lead.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            )
+        else:
+            return self.action_rfq_new()
+
+    def action_rfq_new(self):
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "qoqa_srm.action_lead_rfq_new"
```
Comment (StephaneMangin) [inline srm/models/crm_lead.py]:
```suggestion
            "srm.action_lead_rfq_new"
```

**Review by StephaneMangin — CHANGES_REQUESTED** [review]

File: srm/README.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/111/14.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows the usage of crm module to manage leads coming from suppliers. 
+The flow is similar to CRM. The main change is that leads be generated from customer or supplier request type. 
+For suplier requests leads can be converted in purchases.
```
Comment (StephaneMangin) [inline srm/README.rst:30]:
```suggestion
For supplier requests leads can be converted in purchases.
```

File: srm/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    "author": "Camptocamp, Odoo Community Association (OCA)",
+    "summary": "Use CRM model for suppliers",
+    "license": "AGPL-3",
+    "category": "CRM",
+    "depends": [
+        "account",
+        "crm",
+        "sale_crm",
+        "purchase",
+        "crm_enterprise",
```
Comment (StephaneMangin) [inline srm/__manifest__.py:17]:
```suggestion
```
You can't add this proprietary module in an OCA module.

**Review by StephaneMangin — APPROVED** [review]

File: srm/__manifest__.py (hunk 6 above)
Comment (santostelmo) [reply srm/__manifest__.py:17]: You're right.

**Pull request closed without merging**